### ISSUE TITLE
Floating images for references and backgrounds

### ISSUE THREAD

**Issue by askmeaboutlo0m** [drawpile/Drawpile#1417]
We have a few similar use cases that are pretty common:

1. Users pasting reference image onto the canvas so that they can look at them for reference, pick colors from them etc. This is partially solved by the reference image dock, but it only helps when a single user needs the reference, not when multiple users do.
2. Users pasting background images onto the canvas so that people can draw characters on top of them.
3. Users pasting a rules graphic onto the canvas.

None of these is ideal for size and performance reasons, making it slow to catch up to sessions that use many or large images of this sort. Option 1 is partially solved by the reference image dock, but those image are local-only, so they only help when a single user needs a reference image. For option 2 and 3, it doesn't help at all.

The problem here is really that pasting on a layer is meant for art, so it always preserves every pixel exactly and losslessly. However, for the above use cases, that level of pixel accuracy is excessive. Especially for stuff like photo references or backgrounds, preserving every pixel is totally pointless, since the source image probably isn't lossless to begin with.

We could solve this by adding floating images of some sort. These could let you choose a quality level to bring the size down, using lossy JPEG or WEBP compression without any notable quality reduction, but a huge reduction in size, from several MB to a few hundred KB.

How to put this into the protocol isn't that exciting, since the feature isn't that complicated on that front, but how exactly this is supposed to look in the UI is something that needs ideas and design.

**Comment by amoruwu** [comment]
Annotations can have floating images already which is pretty neat, but you can't do it easily really.

**Comment by askmeaboutlo0m** [comment]
Yeah, annotations being able to contain small-ish images is kind of an accidental feature, but pretty limited by how much data fits into a single annotation, they can't be put into the background and you can't pick colors from them.
